plugins/modules/azure_rm_adapplication_info.py

```diff
@@ -149,7 +229,8 @@ def __init__(self):
             app_id=dict(type='str'),
             object_id=dict(type='str'),
             identifier_uri=dict(type='str'),
-            app_display_name=dict(type='str')
+            app_display_name=dict(type='str'),
+            app_diff=dict(type='list', elements='dict')
```

Comment:
If defined this way, it will be easier to understand!
```suggestion
            app_diff=dict(type='list',
                                   elements='dict',
                                   options=dict(
                                                         app_id=dict(type='str'),
                                                         app_display_name=dict(type='str')
                                   
                                   )
            )
```

Reply:
If I have the configuration of my adapps in a YAML or JSON file, I don't want to edit it additionally. I want to provide it directly to the plugin, not just the "app_display_name" or the "app_id". However, if I provide this as an option, I have to filter the YAML or JSON for these two options, which means another Ansible task in the playbook.

Reply:
But your code only handles these two fields and no other fields?

Reply:
My code need one of these fields, the other fields will be ignored. In the future with this way, it's possible to check the diffrence of any field and to update them.
So I can write the full config for the AD App by code, check it with the app_diff option and create, update or delete the applications with the existing module "azure_rm_adapplication.py".

Reply:
Yes, that is correct. The code only handles these two fields. However, I can provide a YAML or JSON that contains the entire content of multiple ADAPPs in a list without it being blocked. This way, I can directly compare my stored configuration of the ADAPPs without having to parse it again, create a list, and pass the values into a variable